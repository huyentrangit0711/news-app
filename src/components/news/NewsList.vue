<template>
	<div>
		<div
			class="grid grid-cols-3 gap-4"
			:v-show="articles && articles.length > 0"
		>
			<new-item
				v-for="(article, index) in articles"
				:key="index"
				:article="article"
			></new-item>
		</div>
		<div :v-show="loading">
			Loadinbg
		</div>
		test
	</div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';
import NewItem from './NewItem.vue';
export default {
	name: 'NewsList',
	components: {
		NewItem,
	},
	data() {
		return {
			loading: false,
		};
	},
	computed: {
		...mapGetters('news', { articles: 'getArticles' }),
		...mapGetters(['activeTab']),
	},
	methods: {
		...mapActions('news', {
			getTopHeadline: 'getTopHeadLineArticles',
			getCustomNews: 'getNewsByQuery',
		}),
	},
	mounted() {
		if (this.activeTab === 0) {
			this.getTopHeadline();
		} else {
			const defaultQuery = 'animal&bitcoin&apple&earthquake';
			this.getCustomNews(defaultQuery);
		}
	},
};
</script>
