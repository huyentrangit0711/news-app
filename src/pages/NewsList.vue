<template>
	<div class="top-headline-tab">
		<div>
			<h1>Top Headline News</h1>
			<router-view></router-view>
		</div>
	</div>
</template>

<script>
export default {
	name: 'NewsListPage',
};
</script>
