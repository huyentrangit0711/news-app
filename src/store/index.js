import Vue from 'vue';
import Vuex from 'vuex';
import newsModule from './modules/news/index';
Vue.use(Vuex);
const store = new Vuex.Store({
	modules: {
		news: newsModule,
	},
	state() {
		return {
			activeTab: 0,
		};
	},
	mutations: {
		updateActiveTabs(state, payload) {
			state.activeTab = payload.value;
		},
	},
	actions: {
		updateActiveTabs(context, payload) {
			context.commit('updateActiveTabs', payload);
		},
	},
	getters: {
		activeTab(state) {
			return state.activeTab;
		},
	},
});
export default store;
