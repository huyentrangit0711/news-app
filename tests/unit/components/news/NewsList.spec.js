import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import NewsList from '@/components/news/NewsList';
import { actions, getters, mutations } from '../../../../src/store/index';
const localVue = createLocalVue();
localVue.use(Vuex);
describe('News List Component', () => {
	let wrapper;
	let moduleActions;
	let store;
	let news;
	let moduleGetters;
	const createStore = (rootState) => {
		return new Vuex.Store({
			modules: {
				news,
			},
			state: rootState,
			getters,
			actions,
			mutations,
		});
	};
	beforeEach(() => {
		moduleActions = {
			getNewsByQuery: jest.fn(),
			getTopHeadLineArticles: jest.fn(),
		};
		moduleGetters = {
			getArticles: () =>
				jest.fn().mockReturnValue([
					{ id: 1234, title: 'Mock title 1' },
					{ id: 1111, title: 'Mock title 2' },
				]),
			getLoading: () => jest.fn().mockReturnValue(true),
		};
		news = {
			namespaced: true,
			state: {
				articles: [
					{ id: 1234, title: 'Mock title 1' },
					{ id: 1111, title: 'Mock title 2' },
				],
			},
			actions: moduleActions,
			getters: moduleGetters,
		};
	});
	it('Component should render html correctly', async () => {
		store = createStore({
			activeTab: 0,
		});
		wrapper = shallowMount(NewsList, {
			store,
			localVue,
		});
		expect(wrapper).toMatchSnapshot();
	});
	it('Component should render correctly', async () => {
		store = createStore({
			activeTab: 0,
		});
		wrapper = mount(NewsList, {
			store,
			localVue,
		});
		expect(moduleActions.getTopHeadLineArticles).toHaveBeenCalled();
	});
	it('Component should dispatch getCustomNews correctly when activeTab !== 0', async () => {
		store = createStore({
			activeTab: 1,
		});
		wrapper = mount(NewsList, {
			store,
			localVue,
		});
		wrapper.vm.$store.getters.activeTab;
		expect(moduleActions.getNewsByQuery).toHaveBeenCalled();
	});
});
