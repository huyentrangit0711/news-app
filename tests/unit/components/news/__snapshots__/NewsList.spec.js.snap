// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News List Component Component should render html correctly 1`] = `
<div>
  <div class="grid grid-cols-1 lg:grid-cols-3 gap-8 md:grid-cols-2"></div>
  <div>
    <loading-stub></loading-stub>
  </div>
</div>
`;
